api/handlers.go

```diff
@@ -0,0 +1,156 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+// GoCertRouter takes in an environment struct, passes it along to any handlers that will need
+// access to it, then builds and returns it for a server to consume
+func GoCertRouter(env *Environment) http.Handler {
+	router := http.NewServeMux()
+	router.HandleFunc("GET /certificate_requests", GetCertificateRequests(env))
+	router.HandleFunc("POST /certificate_requests", PostCertificateRequest(env))
+	router.HandleFunc("GET /certificate_requests/{id}", GetCertificateRequest(env))
+	router.HandleFunc("DELETE /certificate_requests/{id}", DeleteCertificateRequest(env))
+	router.HandleFunc("POST /certificate_requests/{id}/certificate", PostCertificate(env))
+
+	v1 := http.NewServeMux()
+	v1.HandleFunc("GET /status", HealthCheck)
+	v1.Handle("/api/v1/", http.StripPrefix("/api/v1", router))
+
+	return logging(v1)
```

Comment:
How will this scale when we add more middlewares (Ex. for authentication)? Does returning logging still make sense?

Reply:
we can do something like

`
stack := createMiddlewareStack(
         Logging,
         Authentication,
         Metrics,
         ValidatePermissions,        
)
return stack(v1)
`